### PR TITLE
`azurerm_spring_cloud_gateway` - split to create/update functions

### PR DESCRIPTION
**Pull request by ms-henglu**

### CODE REVIEW

File: internal/services/springcloud/spring_cloud_gateway_resource.go [hunk 1]
```diff
@@ -380,6 +380,96 @@ func resourceSpringCloudGatewayCreateUpdate(d *pluginsdk.ResourceData, meta inte
...
+	springId, err := parse.SpringCloudServiceID(d.Get("spring_cloud_service_id").(string))
+	if err != nil {
+		return err
+	}
+	id := parse.NewSpringCloudGatewayID(subscriptionId, springId.ResourceGroup, springId.SpringName, d.Get("name").(string))
+
+	existing, err := client.Get(ctx, id.ResourceGroup, id.SpringName, id.GatewayName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for existing %s: %+v", id, err)
```
Comment (stephybun) [inline internal/services/springcloud/spring_cloud_gateway_resource.go:398]:
```suggestion
			return fmt.Errorf("retrieving %s: %+v", id, err)
```

File: internal/services/springcloud/spring_cloud_gateway_resource.go [hunk 2]
```diff
@@ -380,6 +380,96 @@ func resourceSpringCloudGatewayCreateUpdate(d *pluginsdk.ResourceData, meta inte
...
+		return fmt.Errorf("retreiving %s: resource was not found", id)
+	}
+
+	if existing.Properties == nil {
+		return fmt.Errorf("retreiving %s: properties was nil", id)
+	}
+	properties := existing.Properties
+
+	if existing.Sku == nil {
+		return fmt.Errorf("retreiving %s: sku was nil", id)
```
Comment (stephybun) [inline internal/services/springcloud/spring_cloud_gateway_resource.go:411]: Retrieving is mispelled here

File: internal/services/springcloud/spring_cloud_gateway_resource.go [hunk 3]
```diff
@@ -380,6 +380,96 @@ func resourceSpringCloudGatewayCreateUpdate(d *pluginsdk.ResourceData, meta inte
...
+	}
+
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.SpringName, id.GatewayName, gatewayResource)
+	if err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation/update of %s: %+v", id, err)
+	}
```
Comment (stephybun) [inline internal/services/springcloud/spring_cloud_gateway_resource.go:467]: Please update the error messages in the create as well 
```suggestion
		return fmt.Errorf("updating %s: %+v", id, err)
	}

	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
		return fmt.Errorf("waiting for update of %s: %+v", id, err)
	}
```

**Review by stephybun — CHANGES_REQUESTED** [review]

**Comment by ms-henglu** [comment]
Hi @stephybun ,

Thanks for the code review! I've updated this PR as suggested, please take another look.

File: internal/services/springcloud/spring_cloud_gateway_resource.go [hunk 4]
```diff
@@ -369,11 +369,101 @@ func resourceSpringCloudGatewayCreateUpdate(d *pluginsdk.ResourceData, meta inte
...
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("retrieving %s: %+v", id, err)
+		}
+	}
+	if utils.ResponseWasNotFound(existing.Response) {
+		return fmt.Errorf("retreiving %s: resource was not found", id)
+	}
+
+	if existing.Properties == nil {
+		return fmt.Errorf("retreiving %s: properties was nil", id)
```
Comment (stephybun) [inline internal/services/springcloud/spring_cloud_gateway_resource.go:406]: still misspelled here

**Review by stephybun — CHANGES_REQUESTED** [review]

**Review by stephybun — APPROVED** [review]
Thanks @ms-henglu LGTM 👍

**Pull request merged**